pkg/stub/imagestreams.go

```diff
@@ -354,3 +293,148 @@ func (h *Handler) clearStreamFromImportError(name string, importError *v1.Config
+
+					}
+					break
+				}
+			}
+		}
+
+		if !anyErrors {
+			if len(is.Spec.Tags) == len(is.Status.Tags) {
+				for _, specTag := range is.Spec.Tags {
+					matched := false
+					for _, statusTag := range is.Status.Tags {
+						logrus.Debugf("checking spec tag %s against status tag %s with num items %d", specTag.Name, statusTag.Tag, len(statusTag.Items))
+						if specTag.Name == statusTag.Tag {
+							// if the latest gens have no errors, see if we got gen match
+							if statusTag.Items != nil {
+								for _, event := range statusTag.Items {
+									if specTag.Generation != nil {
+										logrus.Debugf("checking status tag %d against spec tag %d", event.Generation, *specTag.Generation)
+									}
+									if specTag.Generation != nil &&
+										*specTag.Generation <= event.Generation {
+										logrus.Debugf("got match")
+										matched = true
+										break
+									}
+								}
+							}
+						}
+					}
+					if !matched {
+						pending = true
+						break
+					}
+				}
+			} else {
+				pending = true
+			}
+		}
+	} else {
+		logrus.Debugf("no error/progress checks cause stream %s is skipped", is.Name)
+		// but if skipped, clear out any errors, since we do not care about errors for skipped
+		h.clearStreamFromImportError(is.Name, importError, cfg)
+	}
+
+	processing := cfg.Condition(v1.ImageChangesInProgress)
+
+	logrus.Debugf("pending is %v any errors %v for %s", pending, anyErrors, is.Name)
+
+	// we check for processing == true here as well to avoid churn on relists
+	if !pending {
+		if !anyErrors {
+			h.clearStreamFromImportError(is.Name, importError, cfg)
+		}
+		if processing.Status == corev1.ConditionTrue {
+			now := kapis.Now()
+			// remove this imagestream name, including the space separator, from processing
+			logrus.Debugf("current reason %s ", processing.Reason)
+			replaceOccurs := strings.Contains(processing.Reason, is.Name)
+			if replaceOccurs {
+				processing.Reason = strings.Replace(processing.Reason, is.Name+" ", "", -1)
+				logrus.Debugf("processing reason now %s", processing.Reason)
+				if len(strings.TrimSpace(processing.Reason)) == 0 && processing.Status != corev1.ConditionFalse {
+					logrus.Println("The last in progress imagestream has completed")
+					processing.Status = corev1.ConditionFalse
+					processing.Reason = ""
+				}
+				processing.LastTransitionTime = now
+				processing.LastUpdateTime = now
+				cfg.ConditionUpdate(processing)
+				logrus.Printf("There are no more image imports in flight for imagestream %s", is.Name)
+			}
+		}
+	}
+
+	// clear out error for this stream if there were errors previously but no longer are
+	// think a scheduled import failing then recovering
+	if strings.Contains(importError.Reason, is.Name) && !anyErrors {
+		logrus.Printf("There are no image import errors for %s", is.Name)
+		importError = h.clearStreamFromImportError(is.Name, importError, cfg)
```

Comment:
you don't appear to be doing anything w/ this return value

Reply:
yeah a vestige of an earlier iteration which ultimately was not git pushed ... will clean up